colorcoder/ColorCodeAssociation.java

```diff
@@ -0,0 +1,44 @@
+package colorcoder;
+
+import java.util.Map;
+import java.util.TreeMap;
+
+import colorcoder.utils.NumberValidator;
+
+import static colorcoder.MajorColor.*;
+import static colorcoder.MinorColor.*;
+
+/**
+ * @author {@literal Jayaram Naveenkumar (jayaram.naveenkumar@in.bosch.com)}
+ */
+public final class ColorCodeAssociation {
+
+    private ColorCodeAssociation() {
+    }
+
+    public static PairColor fetchColorFromNumberPair(int numberPair) {
+        if (!NumberValidator.isValid(numberPair)) {
+            throw new IllegalArgumentException("Index number should be in between 1 to 5");
+        }
+        MajorColor majorColor = MajorColor.fetchInstance(numberPair);
+        MinorColor minorColor = MinorColor.fetchInstance(numberPair);
+        return new PairColor(majorColor, minorColor);
+    }
+
+    public static int fetchNumberPairFromColor(
+        MajorColor majorColor,
+        MinorColor minorColor
+    )
+    {
+        return majorColor.fetchIndex() * MINOR_COLOR_SIZE + minorColor.fetchIndex() + 1;
+    }
+
+    public static Map<Integer, String> downloadAllColorCode() {
```

Comment:
is the name 'downloadAllColorCode" is valid name? then what would be the source of download

Reply:
Refactored